### PR TITLE
Add documentation for Variadic args

### PR DESCRIPTION
**Pull request by kevinwilfong**
Summary: Adding documentation for the Variadic args to the documentation for Scalar/Simple functions

Differential Revision: D34082088

### CODE REVIEW

File: velox/docs/develop/scalar-functions.rst [hunk 1]
```diff
@@ -582,6 +582,58 @@ are created and then when they are copied again to the Velox vector. Some work i
...
+
+The last argument to a simple function may be marked "Variadic". This means
+invocations of this function may include 0..N arguments of that type at the end
+of the call.  While not a true type in Velox, "Variadic" can be thought of as a
+syntactic type, and behaves somewhat similarly to Array.
+
+==========  ==============================  =============================
+Velox Type  C++ Argument Type               C++ Actual Argument Type
+==========  ==============================  =============================
+VARIADIC       arg_type<Variadic<E>>              VariadicView<VectorOptionalValueAccessor<VectorReader<E>>>>
```
Comment (laithsakka) [inline velox/docs/develop/scalar-functions.rst:596]: are those spaced needed for rendering (the ones before VariadicView<VectorOptionalValueAccessor<Ve

**Review by laithsakka — APPROVED** [review]

File: velox/docs/develop/scalar-functions.rst [hunk 2]
```diff
@@ -582,6 +582,58 @@ are created and then when they are copied again to the Velox vector. Some work i
...
+
+- size_t size() : return the number of arguments that were passed as part of the "Variadic" type in the function invocation.
+
+- VectorOptionalValueAccessor<arg_type<T>> operator[](size_t index) : access the value of the argument at index.
+
+- VariadicView<T>::Iterator begin() : iterator to the first argument.
+
+- VariadicView<T>::Iterator end() : iterator indicating end of iteration.
+
+- bool mayHaveNulls() : a check on the nullity of the arugments (note this takes time proportional to the number of arguments). When it returns false, there are definitely no nulls, a true does not guarantee null existence.
```
Comment (kevinwilfong) [inline velox/docs/develop/scalar-functions.rst:612]: I realized while writing this that this function is pretty useless (I copied it from ArrayView where it has performance benefits).  I should probably just remove it.

Comment (kevinwilfong) [reply velox/docs/develop/scalar-functions.rst:612]: * I'm referring to mayHaveNulls()

File: velox/docs/develop/scalar-functions.rst (hunk 1 above)
Comment (kevinwilfong) [reply velox/docs/develop/scalar-functions.rst:596]: No, that's my bad, let me fix it, they're supposed to align with the lines above for readability in this file (it doesn't make a difference to rendering)

**Pull request closed without merging**